### PR TITLE
feat: added domain extraction functionality to modal

### PR DESCRIPTION
**Pull request by charith-rc**
### Summary

- Adds frontend logic for the company selection through domain extraction from user email in the add contact modal

### How to test

1.

### Project Checklist

- [x] Changes build without any errors
- [x] Have written adequate test cases
- [x] Done developer testing in
  - [x] Chrome
  - [ ] Firefox
  - [ ] Safari
- [x] Code is formatted with `npm run format`
- [x] Code is linted with `npm run check-lint`
- [x] No unnecessary comments left in code
- [ ] Made corresponding changes to the documentation

#### Other

- [ ] New atomic components added
- [ ] New molecules added
- [ ] New pages(routes) added
- [ ] New dependencies installed

### PR Checklist

- [x] Pull request is raised from the correct source branch
- [x] Pull request is raised to the correct destination branch
- [x] Pull request is raised with correct title
- [x] Pull request is self reviewed
- [x] Pull request is self assigned
- [ ] Suitable pull request status labels are added (`ready-for-code-review`)

### Additional Information
-

### CODE REVIEW

File: frontend/src/community/crm/utils/crmUtil.ts [hunk 1]
```diff
@@ -4,3 +4,24 @@ export const formatValue = (value: NumericValue): string => {
...
+  return items.reduce<{ prioritized: T[]; others: T[] }>(
+    (groups, item) => {
+      const key = priorityIds.includes(Number(item.id)) ? "prioritized" : "others";
+
+      groups[key].push(item);
+      return groups;
+    },
+    { prioritized: [], others: [] }
+  );
+};
```
[the remark this answers is not in the document]
Comment (charith-rc) [reply frontend/src/community/crm/utils/crmUtil.ts]: fixed 8c86bfdc376720de1769794694c71ed89efabae2

File: frontend/src/community/common/components/molecules/SearchableDropdown/SearchableDropdown.tsx [hunk 2]
```diff
@@ -85,6 +87,12 @@ const SearchableDropdown: React.FC<SearchableDropdownProps> = ({
     }
   }, [items.length, activeIndex]);
 
+  useEffect(() => {
+    if (forceOpen) {
+      setIsOpen(true);
+    }
+  }, [forceOpen]);
```
[the remark this answers is not in the document]
Comment (charith-rc) [reply frontend/src/community/common/components/molecules/SearchableDropdown/SearchableDropdown.tsx]: fixed as suggested 303b14eaf2f2b5f4242f49745942e56e379616df

**Review by ThinuwanW — COMMENTED** [review]
## 🤖 Claude Code Review

This PR adds a domain-based company suggestion feature to the Add Contact modal: typing a valid email extracts the domain, searches companies by domain, prioritizes matches in the dropdown, and force-opens the dropdown via a new `forceOpen` prop on SearchableDropdown. The core concern is that domain-matched companies are only re-ordered within the existing lookup results rather than being added, so a matched company outside the default 50-item lookup never surfaces. Smaller issues include a debounced/live value mismatch in the enable flag and a missing unit test for the new utility.

Found **8** new issue(s): 🟡 1 suggestion(s), 🟣 2 nit(s)

File: frontend/src/community/crm/components/molecules/AddContactModalContent/AddContactModalContent.tsx [hunk 3]
```diff
@@ -147,14 +151,35 @@ const AddContactModalContent: React.FC = () => {
...
+    const items =
+      companyLookupData?.items?.map((company) => ({
+        id: String(company.id),
+        content: company.name
+      })) ?? [];
+
+    const domainCompanyIds =
+      domainSearchData?.companies?.map((company) => company.id) ?? [];
+
+    const { prioritized, others } = prioritizeListIds(items, domainCompanyIds);
```
Comment (ThinuwanW) [inline frontend/src/community/crm/components/molecules/AddContactModalContent/AddContactModalContent.tsx]: 🤖 **Claude** · 💬 **Note**

`prioritizeListIds` only re-orders companies that already exist in `companyLookupData.items`; it never adds the domain-matched companies. When the company search box is empty, `useGetCompanyLookup` runs with an empty keyword and returns at most DEFAULT_LOOKUP_PAGE_SIZE (50) companies. For a tenant with >50 companies, the company that actually matches the email domain may not be in that first 50 and will therefore never appear in the dropdown — defeating the primary purpose of the feature. Fix: merge `domainSearchData.companies` into the items list (mapping to SearchableDropdownItem and de-duplicating by id) before/while prioritizing, rather than using domain results purely as a sort key.

File: frontend/src/community/crm/components/molecules/AddContactModalContent/AddContactModalContent.tsx [hunk 4]
```diff
@@ -147,14 +151,35 @@ const AddContactModalContent: React.FC = () => {
     submitForm
   } = formik;
 
+  const extractedDomain = extractDomainFromEmail(values.email);
+
+  const debouncedDomain = useDebounce(extractedDomain, SEARCH_DEBOUNCE_DELAY);
+
+  const isDomainSearchEnabled =
```
Comment (ThinuwanW) [inline frontend/src/community/crm/components/molecules/AddContactModalContent/AddContactModalContent.tsx:161]: 🤖 **Claude** · 💬 **Note**

`isDomainSearchEnabled` mixes a debounced value with a live one: `debouncedDomain.length > 0` (debounced) is combined with `emailPattern().test(values.email)` (un-debounced). Because `debouncedDomain` lags `values.email` by SEARCH_DEBOUNCE_DELAY, the flag can briefly toggle on a stale/in-progress email, which also drives `forceOpen` and can cause the dropdown to flicker open. Recommend testing the pattern against a consistent source — e.g. derive both domain and the validity check from the same debounced email value.

File: frontend/src/community/crm/utils/crmUtil.ts [hunk 5]
```diff
@@ -4,3 +4,24 @@ export const formatValue = (value: NumericValue): string => {
...
+  id: number | string;
+}
+
+export const prioritizeListIds = <T extends Identifiable>(
+  items: T[],
+  priorityIds: number[]
+): { prioritized: T[]; others: T[] } => {
+  return items.reduce<{ prioritized: T[]; others: T[] }>(
+    (groups, item) => {
+      const key = priorityIds.includes(Number(item.id))
```
Comment (ThinuwanW) [inline frontend/src/community/crm/utils/crmUtil.ts]: 🤖 **Claude** · 🟣 **Nit**

`prioritizeListIds<T extends Identifiable>` allows `id: number | string`, but `priorityIds` is typed `number[]` and matching is done via `priorityIds.includes(Number(item.id))`. If this generic util is ever reused with non-numeric string ids, `Number(item.id)` yields `NaN`, which silently never matches and pushes everything to `others`. Consider normalizing both sides to strings (e.g. compare `priorityIds.map(String).includes(String(item.id))`) or constraining the types so the contract is explicit. (Current crm usage with numeric ids is fine, hence nit.)

File: frontend/src/community/crm/utils/crmUtil.ts [hunk 6]
```diff
@@ -4,3 +4,24 @@ export const formatValue = (value: NumericValue): string => {
   if (value == null || value === "") return "-";
   return `$${Number.parseFloat(value).toFixed(2)}`;
 };
+
+interface Identifiable {
+  id: number | string;
+}
+
+export const prioritizeListIds = <T extends Identifiable>(
```
Comment (ThinuwanW) [inline frontend/src/community/crm/utils/crmUtil.ts]: 🤖 **Claude** · 🟡 **Suggestion**

New exported pure utility `prioritizeListIds` has no corresponding `.test.ts`. Per the testing guideline, add unit tests covering the prioritized/others split, empty `priorityIds`, and items with no matching id.

File: frontend/src/community/common/components/molecules/SearchableDropdown/SearchableDropdown.tsx [hunk 7]
```diff
@@ -85,6 +87,15 @@ const SearchableDropdown: React.FC<SearchableDropdownProps> = ({
     }
   }, [items.length, activeIndex]);
 
+  useEffect(() => {
```
Comment (ThinuwanW) [inline frontend/src/community/common/components/molecules/SearchableDropdown/SearchableDropdown.tsx]: 🤖 **Claude** · 🟣 **Nit**

The new `forceOpen` effect forces `isOpen=false` (and clears `activeIndex`) whenever `forceOpen` transitions to false. If a user has typed text in the company search field (dropdown open via `handleInputChange`) and then edits the email so it becomes invalid, `forceOpen` flips to false and this effect closes the dropdown out from under them even though they were actively searching companies. Consider only closing on the forceOpen→false transition when the input itself is empty, or gating the close so it doesn't override user-driven open state.

File: frontend/src/community/crm/components/molecules/AddContactModalContent/AddContactModalContent.tsx [hunk 8]
```diff
@@ -147,14 +151,35 @@ const AddContactModalContent: React.FC = () => {
...
+    debouncedDomain.length > 0 && emailPattern().test(values.email);
```
Comment (ThinuwanW) [inline frontend/src/community/crm/components/molecules/AddContactModalContent/AddContactModalContent.tsx]: 🤖 **Claude** · 💬 **Note**

The domain-search gate validates the email with `emailPattern()` (from regexPatterns), but the form's Yup schema (addContactValidations -> `isValidEmail()`) validates the same field with a DIFFERENT regex. The two patterns are not equivalent (e.g. `isValidEmail()` permits quoted local parts and bracketed IP-literal domains; `emailPattern()` does not). This means the domain search can fail to fire for emails the form accepts, or fire for ones it rejects. Reuse a single source of truth — call `isValidEmail()` here (or extract one shared helper) so domain-search gating and form validation can never drift apart. This is distinct from the debounced/live timing issue: it is about two divergent regexes for the same field.

File: frontend/src/community/common/components/molecules/SearchableDropdown/SearchableDropdown.tsx [hunk 9]
```diff
@@ -85,6 +87,15 @@ const SearchableDropdown: React.FC<SearchableDropdownProps> = ({
...
+    if (forceOpen) {
+      setIsOpen(true);
+    } else {
+      setIsOpen(false);
```
Comment (ThinuwanW) [inline frontend/src/community/common/components/molecules/SearchableDropdown/SearchableDropdown.tsx]: 🤖 **Claude** · 💬 **Note**

The new force-close branch (`forceOpen` -> false) calls `setIsOpen(false)` / `setActiveIndex(null)` directly but does NOT invoke `onClose?.()`, unlike every other close path (`handleClose`, Escape, Tab, select all route through `handleClose` which calls `onClose`). As a result, parent cleanup wired to onClose is skipped on a force-close — e.g. in AddContactModalContent `onClose={() => setCompanySearch("")}` never runs, so the company search text is left stale when the email becomes invalid and the dropdown auto-closes. Route this branch through `handleClose()` (or call `onClose?.()`) so the close contract is honored consistently. (Distinct from the first-pass note about clearing typed/active state — this is specifically the missing onClose invocation.)

File: frontend/src/community/crm/components/molecules/AddContactModalContent/AddContactModalContent.tsx [hunk 10]
```diff
@@ -147,14 +151,35 @@ const AddContactModalContent: React.FC = () => {
...
+
+  const { data: domainSearchData } = useSearchCompaniesByDomain(
```
Comment (ThinuwanW) [inline frontend/src/community/crm/components/molecules/AddContactModalContent/AddContactModalContent.tsx:164]: 🤖 **Claude** · 💬 **Note**

The `useSearchCompaniesByDomain` query result only destructures `data`; there is no handling of its error/loading state. If the domain search request fails, it fails silently and the company dropdown is still force-opened (via `forceOpen={isDomainSearchEnabled}`) showing the generic lookup list, with no indication the enrichment failed. Since this is a background enrichment a toast may be undesirable, but at minimum consider not forcing the dropdown open until `domainSearchData` has resolved successfully, so the auto-open is tied to actual domain results rather than merely a valid email. Flagging as a suggestion for the developer to decide the intended UX.

Comment (charith-rc) [reply frontend/src/community/crm/components/molecules/AddContactModalContent/AddContactModalContent.tsx:164]: changed behaviour to open on focus https://github.com/SkappHQ/skapp/pull/1478/commits/76542925a6776ffe1369438eb0f5dfdb7139c6b4

File: frontend/src/community/common/components/molecules/SearchableDropdown/SearchableDropdown.tsx (hunk 9 above)
Comment (charith-rc) [reply frontend/src/community/common/components/molecules/SearchableDropdown/SearchableDropdown.tsx]: switched the force behaviour to on focus https://github.com/SkappHQ/skapp/pull/1478/commits/76542925a6776ffe1369438eb0f5dfdb7139c6b4

File: frontend/src/community/common/components/molecules/SearchableDropdown/SearchableDropdown.tsx (hunk 7 above)
Comment (charith-rc) [reply frontend/src/community/common/components/molecules/SearchableDropdown/SearchableDropdown.tsx]: removed force open behaviour, now only opens on focus https://github.com/SkappHQ/skapp/pull/1478/commits/76542925a6776ffe1369438eb0f5dfdb7139c6b4

File: frontend/src/community/crm/utils/crmUtil.ts (hunk 6 above)
Comment (charith-rc) [reply frontend/src/community/crm/utils/crmUtil.ts]: no need

File: frontend/src/community/crm/components/molecules/AddContactModalContent/AddContactModalContent.tsx (hunk 8 above)
Comment (charith-rc) [reply frontend/src/community/crm/components/molecules/AddContactModalContent/AddContactModalContent.tsx]: replaced with isValidEmail() pattern https://github.com/SkappHQ/skapp/pull/1478/commits/9243fd3f3571702601db18992aedb83bd94e14db

File: frontend/src/community/crm/utils/crmUtil.ts (hunk 5 above)
Comment (charith-rc) [reply frontend/src/community/crm/utils/crmUtil.ts]: reworked to compare strings as suggested https://github.com/SkappHQ/skapp/pull/1478/commits/9243fd3f3571702601db18992aedb83bd94e14db

File: frontend/src/community/crm/components/molecules/AddContactModalContent/AddContactModalContent.tsx (hunk 4 above)
Comment (charith-rc) [reply frontend/src/community/crm/components/molecules/AddContactModalContent/AddContactModalContent.tsx:161]: reworked to drive both from debounced value https://github.com/SkappHQ/skapp/pull/1478/commits/9243fd3f3571702601db18992aedb83bd94e14db

File: frontend/src/community/crm/components/molecules/AddContactModalContent/AddContactModalContent.tsx (hunk 3 above)
Comment (charith-rc) [reply frontend/src/community/crm/components/molecules/AddContactModalContent/AddContactModalContent.tsx]: handled missing data adding https://github.com/SkappHQ/skapp/pull/1478/commits/9243fd3f3571702601db18992aedb83bd94e14db

File: frontend/src/community/common/components/molecules/SearchableDropdown/SearchableDropdown.tsx [hunk 11]
```diff
@@ -55,6 +55,7 @@ export interface SearchableDropdownProps {
   variant?: "sm" | "md" | "lg";
   positionStrategy?: "absolute" | "fixed";
   onClose?: () => void;
+  openOnFocus?: boolean;
 }
 
 const SearchableDropdown: React.FC<SearchableDropdownProps> = ({
```
Comment (anusham-rc) [inline frontend/src/community/common/components/molecules/SearchableDropdown/SearchableDropdown.tsx]: remove React

File: frontend/src/community/crm/components/molecules/AddContactModalContent/AddContactModalContent.tsx [hunk 12]
```diff
@@ -256,6 +299,10 @@ const AddContactModalContent: React.FC = () => {
               </p>
             )
           }
+          openOnFocus={
+            isDomainSearchEnabled &&
+            (domainSearchData?.companies?.length ?? 0) > 0
```
Comment (anusham-rc) [inline frontend/src/community/crm/components/molecules/AddContactModalContent/AddContactModalContent.tsx]: check this

Comment (charith-rc) [reply frontend/src/community/crm/components/molecules/AddContactModalContent/AddContactModalContent.tsx]: rewrote to be more readable https://github.com/SkappHQ/skapp/pull/1478/commits/003cc9195caa0b0e964b29c9c4f5173d7a371650

File: frontend/src/community/common/components/molecules/SearchableDropdown/SearchableDropdown.tsx (hunk 11 above)
Comment (charith-rc) [reply frontend/src/community/common/components/molecules/SearchableDropdown/SearchableDropdown.tsx]: removed https://github.com/SkappHQ/skapp/pull/1478/commits/003cc9195caa0b0e964b29c9c4f5173d7a371650

**Review by anusham-rc — DISMISSED** [review]

**Review by sanjula-rtc — APPROVED** [review]

**Review by anusham-rc — APPROVED** [review]

**Review by thusala — APPROVED** [review]

**Review by anusham-rc — APPROVED** [review]

**Review by ThinuwanW — APPROVED** [review]
